lib/runtime-core/src/module.rs

```diff
@@ -100,8 +100,12 @@ impl Module {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn instantiate(&self, import_object: &ImportObject) -> error::Result<Instance> {
-        Instance::new(Arc::clone(&self.inner), import_object)
```

Comment:
We should probably keep this public signature around for backwards compatability: `pub fn instantiate(&self, import_object: &ImportObject) -> error::Result<Instance> {`
I'm guessing we may have more configuration later on, so maybe we could have a `Config` struct to hold the `symbol_map`:
```
  pub fn instantiate_with_config(
        &self,
        import_object: &ImportObject,
        config: &Config,
    )
```